components/brave_wallet/browser/blockchain_registry.cc

```diff
@@ -308,4 +496,64 @@ bool BlockchainRegistry::IsOfacAddress(const std::string& address) {
   return base::Contains(ofac_addresses_, base::ToLowerASCII(address));
 }
 
+void BlockchainRegistry::ParseLists(const base::FilePath& install_dir,
+                                    base::OnceClosure callback) {
+  if (install_dir.empty()) {
+    std::move(callback).Run();
+    return;
+  }
+
+  // ResolveAbsolutePath needs to be posted due to I/O operation.
+  // base::Unretained(this) is safe here due to NoDestructor.
+  sequenced_task_runner_->PostTaskAndReplyWithResult(
+      FROM_HERE, base::BindOnce(&ResolveAbsolutePath, install_dir),
+      base::BindOnce(&BlockchainRegistry::DoParseLists, base::Unretained(this),
+                     std::move(callback)));
+}
+
+void BlockchainRegistry::DoParseLists(
+    base::OnceClosure callback,
+    const base::FilePath& absolute_install_dir) {
+  if (absolute_install_dir.empty()) {
+    std::move(callback).Run();
+    return;
+  }
+
+  // These parsing function needs to be posted due to reading JSON files.
+  sequenced_task_runner_->PostTaskAndReply(
+      FROM_HERE,
+      base::BindOnce(
+          [](const base::FilePath& absolute_install_dir) {
+            ParseCoingeckoIdsMapAndUpdateRegistry(absolute_install_dir);
+            ParseTokenListAndUpdateRegistry(absolute_install_dir);
+            ParseChainListAndUpdateRegistry(absolute_install_dir);
+            ParseDappListsAndUpdateRegistry(absolute_install_dir);
+            ParseOnRampListsAndUpdateRegistry(absolute_install_dir);
+            ParseOfacAddressesListsAndUpdateRegistry(absolute_install_dir);
+          },
+          absolute_install_dir),
+      std::move(callback));
```

Comment:
1) We don't need two calls on MayBlock runner. Just one which resolves path and parses jsons.
2) So instead `sequenced_task_runner_` just `base::ThreadPool::PostTaskAndReplyWithResult(FROM_HERE, {base::MayBlock(), ...`
3) That blocking operation should produce an 'internal state' of BlockchainRegistry:
  ```
    CoingeckoIdsMap coingecko_ids_map_;
    TokenListMap token_list_map_;
    etc...
  ```
  and return it as `...ReplyWithResult`
4) `BlockchainRegistry::GetInstance()` object is now(and actually was) accessed from different threads and should be fixed by (3). Also makes sense to add DCHECK_CALLED_ON_VALID_SEQUENCE for instance of BlockchainRegistry::GetInstance() internal static.

Reply:
Fixed by https://github.com/brave/brave-core/pull/20741/commits/f1e4487746ffded61f315d30f8555d3b8f95aff7.